python/tvm/relay/frontend/pytorch.py

```diff
@@ -1735,14 +1735,19 @@ def pad(inputs, input_types):
             paddings = [paddings[i : i + 2] for i in range(0, len(paddings), 2)]
 
             const_paddings = []
+            non_zero_found = False
             for pad in paddings:
                 const_paddings.append([])
                 for p in pad:
                     if not isinstance(p, int):
                         p = int(_infer_value(p, {}).numpy())
                     const_paddings[-1].append(p)
+                    if p != 0:
+                        non_zero_found = True
 
-            if mode == "constant":
+            if not non_zero_found:
+                return data
```

Comment:
This is a minor optimization but it non-trivially helped performance on the DETR model. @comaniac

Reply:
Hmm interesting. I didn't notice that we may have pad ops that actually pad nothing.